### PR TITLE
New Flavor: MTP

### PR DESCRIPTION
**Pull request by matiaspakua**
## Description
adding a new flavor, a MTP = Master Test Plan.

## Type of change
Please check options that are relevants.

- [X] New content
- [ ] Website changes (fix or new feature)
- [ ] This change requires a documentation update

## Checklist:
Please check options that are relevants.

- [X] My content follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have contributed on the `trello` board as well

### CODE REVIEW

File: content/Flavours/mtp.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+A good way to start is with the integrated Wiki on platforms as Gitlab, Github (to be stored in the same repository as the source code) o even in a shared folder in OneDrive, GoogleDrive or any useful tool used in the organization.
+
+## Example Template
+
+The following is a example structure:
+
+```markdown
+1. Introduction
+1.1. Proyect description
```
Comment (lenntt) [inline content/Flavours/mtp.md]: Typo: "Project"

File: content/Flavours/mtp.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ - Defines a clear strategy to ensure quality during all the SDLC.
+ - Defines the testing process, stages and flows.
+ - Defines the set of tools needed to flow the process.
+ - Define the flow of work for the development of the product.
+ - Ensures the Quality of the prodcut.
+
+## How to
+As with the ADR or SAD, a MTP need to be simple, well defined, available for everyone, need to be in constant change and improvement.
+
+A good way to start is with the integrated Wiki on platforms as Gitlab, Github (to be stored in the same repository as the source code) o even in a shared folder in OneDrive, GoogleDrive or any useful tool used in the organization.
```
Comment (lenntt) [inline content/Flavours/mtp.md]: Typo: o -> "or"

File: content/Flavours/mtp.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+## Example Template
+
+The following is a example structure:
+
+```markdown
+1. Introduction
+1.1. Proyect description
+1.2. Product Description
+2. Scope
+2.1. SUT - Software Under Test Overwie
```
Comment (lenntt) [inline content/Flavours/mtp.md]: Typo: "Overview"

File: content/Flavours/mtp.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+# MTP - Master Test Plan
+## Description
+A MTP is a documents that write down the Testing Strategy for a software product. Includes the Process, methodology, tools, criteria and all the general aspects associated with the Quality Assurance.
+
+## Why?
+Define the Testing "strategy" is a key aspect in software Engineering. A MTP will define the Quality of the software, not only at proper functionality level, but beyond, event at Customer Value level and profit.
+
+At a technical view, a MTP helps the Organization reducing futuro cost, failures, impruvement of confident and creates a base of trusth among the people that develop products.
```
Comment (lenntt) [inline content/Flavours/mtp.md]: typos: "future" and "improvement of confidence" and "trust"

File: content/Flavours/mtp.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Problems
+A MTP help on:
+ - Defines a clear strategy to ensure quality during all the SDLC.
+ - Defines the testing process, stages and flows.
+ - Defines the set of tools needed to flow the process.
+ - Define the flow of work for the development of the product.
+ - Ensures the Quality of the prodcut.
```
Comment (lenntt) [inline content/Flavours/mtp.md]: product

File: content/Flavours/mtp.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+## Constraint
+ - Start as soon as possible.
+ - Always is best to start now.
+ - Work on the MTP in a iterative and incremental manner.
+ - Always update the MTP when a change is made.
+
+> Keep it Simple: avoid long text descriptions, make visual explanation instead.
+
+## Resources
+ - [MTP Examples](http://www1.scdhhs.gov/tcoe/worksoft_sdlc/workproducts/master_test_plan_BC36C914.html)
```
Comment (lenntt) [inline content/Flavours/mtp.md:73]: Link doesn't work for me

**Review by lenntt — COMMENTED** [review]
I was just passing by (not a contributor (yet)) and ran against this PR and identified some typos and broken link.

I would agree making a testing strategy explicit is useful and helps maturing a team/product. 
And typically help teams with defining things described here. Yet, the structure feels like something I would not do this way.
Personally, I would prefer the word 'quality' in the title over 'test', as QA isn't limited to testing. I tend to start with a vision to share (often a context-specific version of shift-left'), and then have a visualisation of the (automated) delivery flow, linking steps and gaps. E.g. coding guidelines and review processes are also in there.
This helps identifying bottlenecks and re-evaluation as well keeping the focus on devops capabilities.

File: content/Flavours/mtp.md (hunk 1 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md]: fixed

File: content/Flavours/mtp.md (hunk 2 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md]: fixed

File: content/Flavours/mtp.md (hunk 3 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md]: fix

File: content/Flavours/mtp.md (hunk 4 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md]: fix

File: content/Flavours/mtp.md (hunk 5 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md]: fix

File: content/Flavours/mtp.md (hunk 6 above)
Comment (matiaspakua) [reply content/Flavours/mtp.md:73]: maybe is because is an unsecure (http) URL?  because works for me.

**Comment by matiaspakua** [comment]
> I was just passing by (not a contributor (yet)) and ran against this PR and identified some typos and broken link.
> 
> I would agree making a testing strategy explicit is useful and helps maturing a team/product. And typically help teams with defining things described here. Yet, the structure feels like something I would not do this way. Personally, I would prefer the word 'quality' in the title over 'test', as QA isn't limited to testing. I tend to start with a vision to share (often a context-specific version of shift-left'), and then have a visualisation of the (automated) delivery flow, linking steps and gaps. E.g. coding guidelines and review processes are also in there. This helps identifying bottlenecks and re-evaluation as well keeping the focus on devops capabilities.

>>>>
Interesting comment, I like it. Is true, Quality is far more than testing, in that case, the practice could be called: Master Quality Plan (MQP) and that deserve come back to this PR. I will try to re-think that. Thanks.

File: content/Flavours/mtp.md (hunk 6 above)
Comment (lenntt) [reply content/Flavours/mtp.md:73]: image
That could be it, I'd avoid unsecure URLs.
(I'm on Chrome on macOS)

**Comment by lenntt** [comment]
> > I was just passing by (not a contributor (yet)) and ran against this PR and identified some typos and broken link.
> > I would agree making a testing strategy explicit is useful and helps maturing a team/product. And typically help teams with defining things described here. Yet, the structure feels like something I would not do this way. Personally, I would prefer the word 'quality' in the title over 'test', as QA isn't limited to testing. I tend to start with a vision to share (often a context-specific version of shift-left'), and then have a visualisation of the (automated) delivery flow, linking steps and gaps. E.g. coding guidelines and review processes are also in there. This helps identifying bottlenecks and re-evaluation as well keeping the focus on devops capabilities.
> 
> > > > >
> 
> Interesting comment, I like it. Is true, Quality is far more than testing, in that case, the practice could be called: Master Quality Plan (MQP) and that deserve come back to this PR. I will try to re-think that. Thanks.

Would like to help here, maybe we can start with a good name. Lets break it down:

- 'Master' I think this is a word to avoid nowadays. 'main' would be al alternative, but I wonder if the word adds anything? Are there non-master/main plans as well?
- Are we describing a Plan, a Strategy or a combination? Or maybe it is an instance of Architecture decisions?
- There's a lot we _can_ write down on quality. what is the thing we really want to get out of it?

**Comment by Tr00d** [comment]
Thanks @matiaspakua for contributing to Xtrem-TDD, we really appreciate your effort.
I'd like to point out a few things to initiate new discussions:
- We noted several typos in both this PR and the previous one. Try to review the content first before publishing the PR, it will help a lot to get a smooth flow.
- I think it would be interesting to discuss altogether before putting new content in. We have like 60-70ish topics we'd like to put in.
   - They're all in the Trello board
   - There's also a Discord server, feel free to ping us so that we can invite you in.
- I feel there are a lot of similarities between this page and the SAD one (for example, the `constraint` region seems to be identical). So far, the content we created is oriented towards problem-solving. So maybe we can discuss together to find what problem we're helping others to solve here? If the idea is to help preparing a test plan for a software, I'm open to it but maybe the content needs to be expanded to be really usable per-se.

Don't hesitate to give me your feedback, always a pleasure to talk 😃

**Comment by matiaspakua** [comment]
Hello, Thanks for the feedback . Maybe this PR can be closed without merge and I take your feedback to work on it on my fork. If there is a Discord or a Trello  board I am happy to contribute what whereber I can. I like vey much the initiative. Feel free to contact me and we can sync on that.

**Comment by Tr00d** [comment]
Sure, there you go: https://discord.gg/nHQUhvtB

**Comment by matiaspakua** [comment]
to re-work on the "flavor"

**Pull request closed without merging**